src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -125,10 +126,16 @@ class GcsActor {
   /// Get the mutable ActorTableData of this actor.
   rpc::ActorTableData *GetMutableActorTableData();
 
+  std::shared_ptr<GcsActorWorkerAssignment> GetActorWorkerAssignment() const;
+
+  void SetActorWorkerAssignment(std::shared_ptr<GcsActorWorkerAssignment> assignment_ptr);
+
  private:
   /// The actor meta data which contains the task specification as well as the state of
   /// the gcs actor and so on (see gcs.proto).
   rpc::ActorTableData actor_table_data_;
+  // TODO(Chong-Li): Considering shared assignments, this pointer would be moved out.
+  std::shared_ptr<GcsActorWorkerAssignment> assignment_ptr_ = nullptr;
```

Comment:
Could you give a detailed explanation about why we need it to be a shared_ptr? Could unique_ptr be good enough?

Reply:
With `class GcsActorWorkerAssignment` declared above, using `unique_ptr` would give a `Pimpl` comping error. It could be fixed, but this pointer is just a workaround. It'll be removed out in the next PR when we consider shared worker processes.